clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -830,77 +830,20 @@ class KernelObjVisitor {
-    if (Util::isSyclAccessorType(ElementTy))
-      KF_FOR_EACH(handleSyclAccessorType, ArrayField, ElementTy);
-    else if (Util::isSyclStreamType(ElementTy))
-      KF_FOR_EACH(handleSyclStreamType, ArrayField, ElementTy);
-    else if (Util::isSyclSamplerType(ElementTy))
-      KF_FOR_EACH(handleSyclSamplerType, ArrayField, ElementTy);
-    else if (Util::isSyclHalfType(ElementTy))
-      KF_FOR_EACH(handleSyclHalfType, ArrayField, ElementTy);
-    else if (ElementTy->isStructureOrClassType())
-      VisitRecord(Owner, ArrayField, ElementTy->getAsCXXRecordDecl(),
-                  handlers...);
-    else if (ElementTy->isUnionType())
-      VisitUnion(Owner, ArrayField, ElementTy->getAsCXXRecordDecl(),
-                 handlers...);
-    else if (ElementTy->isArrayType())
-      VisitArrayElements(ArrayField, ElementTy, handlers...);
-    else if (ElementTy->isScalarType())
-      KF_FOR_EACH(handleScalarType, ArrayField, ElementTy);
-  }
-
-  template <typename... Handlers>
-  void VisitFirstElement(CXXRecordDecl *Owner, FieldDecl *ArrayField,
-                         QualType ElementTy, Handlers &... handlers) {
-    VisitElementImpl(Owner, ArrayField, ElementTy, handlers...);
-  }
-
-  template <typename... Handlers>
-  void VisitNthElement(CXXRecordDecl *Owner, FieldDecl *ArrayField,
-                       QualType ElementTy, Handlers &... handlers);
-
-  template <typename... Handlers>
-  void VisitArrayElements(FieldDecl *FD, QualType FieldTy,
-                          Handlers &... handlers) {
-    const ConstantArrayType *CAT =
-        SemaRef.getASTContext().getAsConstantArrayType(FieldTy);
-    assert(CAT && "Should only be called on constant-size array.");
-    QualType ET = CAT->getElementType();
-    int64_t ElemCount = CAT->getSize().getSExtValue();
-    std::initializer_list<int>{(handlers.enterArray(), 0)...};
-
-    assert(ElemCount > 0 && "SYCL prohibits 0 sized arrays");
-    VisitFirstElement(nullptr, FD, ET, handlers...);
-    (void)std::initializer_list<int>{(handlers.nextElement(ET, 1), 0)...};
-
-    for (int64_t Count = 1; Count < ElemCount; Count++) {
-      VisitNthElement(nullptr, FD, ET, handlers...);
-      (void)std::initializer_list<int>{
-          (handlers.nextElement(ET, Count + 1), 0)...};
-    }
-
-    (void)std::initializer_list<int>{
-        (handlers.leaveArray(FD, ET, ElemCount), 0)...};
-  }
-
   // Parent contains the FieldDecl or CXXBaseSpecifier that was used to enter
   // the Wrapper structure that we're currently visiting. Owner is the parent
   // type (which doesn't exist in cases where it is a FieldDecl in the
   // 'root'), and Wrapper is the current struct being unwrapped.
   template <typename ParentTy, typename... Handlers>
   void VisitRecord(const CXXRecordDecl *Owner, ParentTy &Parent,
-                   const CXXRecordDecl *Wrapper, Handlers &... handlers) {
+                   const CXXRecordDecl *Wrapper, QualType RecordTy,
+                   Handlers &... handlers) {
```

Comment:
I guess "handlers" also should start with upper case letter...

Reply:
Note that I couldn't just capitalize Handlers, since that was taken by the pack.  I changed the pack to HanderTys, and the variables to Handlers.  I did this throughout the visitor type.

Reply:
Yeah, that is why I didn't make this comment in form of dummy suggestion.